packages/utils/src/managers/ZIndexManager.tsx

```diff
@@ -0,0 +1,39 @@
+import React, { FC, useEffect, useState } from 'react';
+
+
+export interface IZIndexManager {
+  ZIndexContext: React.Context<number>;
+  Provider: FC;
+}
+
+export const getMaxZIndex = (): number | undefined => Array.from(document.querySelectorAll('body *'))
+  .map((node: Element) => parseFloat(window.getComputedStyle(node).zIndex))
+  .filter((zIndex: number) => !Number.isNaN(zIndex))
+  .sort((a, b) => a - b)
+  .pop();
+
+export const ZIndexManager = (): IZIndexManager => {
```

Comment:
I am not sure about ZIndexManager. Why not just export ZIndexContext and ZIndexProvider?

Reply:
Because there may be a different context in different components. E.g.
The modal component has a high priority and it is always higher located on the page
and all other components under the modal window